### PR TITLE
feat: Use CometPlugin as main entrypoint

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Closes #.

### CODE REVIEW

File: spark/src/main/scala/org/apache/spark/Plugins.scala [hunk 1]
```diff
@@ -44,6 +45,20 @@ class CometDriverPlugin extends DriverPlugin with Logging with ShimCometDriverPl
...
+    if (sc.conf.contains(extensionKey)) {
+      val extensions = sc.conf.get(extensionKey)
+      if (!extensions.split(",").map(_.trim).contains(extensionClass)) {
+        sc.conf.set(extensionKey, s"$extensions,$extensionClass")
+      } else {
+        sc.conf.set(extensionKey, extensionClass)
+      }
+    } else {
+      sc.conf.set(extensionKey, extensionClass)
+    }
```
Comment (comphead) [inline spark/src/main/scala/org/apache/spark/Plugins.scala:60]: its probably would be good to log the overriden value for `spark.sql.extensions` ?

**Review by comphead — COMMENTED** [review]
Great idea of having CometPlugin as main entry point

File: spark/src/main/scala/org/apache/spark/Plugins.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (edmondop) [reply spark/src/main/scala/org/apache/spark/Plugins.scala:60]: The else clause it's also suspicious, are we sure if there are multiple extensions with the key we want to override everything? I suspect the code doesn't need an else branch in reality

[the remark this answers is not in the document]
Comment (andygrove) [reply spark/src/main/scala/org/apache/spark/Plugins.scala]: Thanks for the review @edmondop. You are correct that the original code was not correct. I have added a unit test and fixed the issue. It is close to your suggestion.

**Review by comphead — APPROVED** [review]
lgtm thanks @andygrove 
the formatting needs to be fixed

**Pull request merged**